doc/LibcurlTransportAdapter.md

```diff
@@ -0,0 +1,162 @@
+
+## Libcurl Transport Adapter
+
+The azure-core-cpp library provides a transport adapter implemented with libcurl. The next paragraphs will mention some of the limitations of libcurl and the code that is part of the LibcurlTransportAdatper  (or LTA) that satisfy the expectations for an SDK client.
+
+### Easy handle
+
+Let's start by talking about libcurl fundamentals. Libcurl is written in C, as such it is fast and practical. It provides two main modes of operation, the easy handle and the multi handle. The easy handle is the starting point for both. The multi handle is nothing but a group of easy handles.
+ 
+An easy handle represents a network request to be performed. For a developer, the typical implementation flow is:
+
+1.	Create an easy handle (as a raw pointer).
+2.	Ask libcurl to init the handle. Libcurl handles memory allocation.
+3.	Configure handle according to what needs to be requested to server.
+4.	Tell libcurl to take the easy handle and perform the request.
+ 
+As simple as that, libcurl is putting away a ton of complexity from developer's hands. If step 3 was done correctly, libcurl would ask the Operating System for a network socket to speak to the network. It will establish a connection. It will even take care of securing the connection with TLS when requested. It will talk to a server following HTTP spec and it will receive and parse a response from the network. It sounds magical! However, it is not enough for the Azure SDK requirements.
+
+### Azure SDK requirements
+
+From all the operations supported by Azure services, there's one that can't be fully completed by following the libcurl typical flow mentioned before. It is the download operation from the Azure Storage service which requires something else. When the size of the downloaded item is bigger than the amount of memory in the system, for example, downloading 1Tb from an Android device with only 8GB of memory, or from a Windows PC with 32 GB of memory, there should be no complications for an SDK client. However, based on the typical libcurl flow, during step 4, libcurl blocks the application until all the bytes from the server have been received, and that's not acceptable from the SDK perspective.
+
+#### Stream response on demand
+
+The expectation when using the SDK client to download bytes is to enable an application to read any amount of data from the network, on demand. An application must be able to read any amount of data from the network, then, it must be able to do something with that data and decide when to read some more. 
```

Comment:
Too many ',' (commas). Possibly "An application must be able to read any amount of data from the network, do something with that data and then decide when to read some more" (or something like that).